### PR TITLE
warn_error_options docs

### LINKED ISSUE #2687: New global config: `--warn-error-options`
### Contributions

- [X] I have read the contribution docs, and understand what's expected of me.

### Link to the page on docs.getdbt.com requiring updates

https://docs.getdbt.com/reference/global-configs#warnings-as-errors

### What part(s) of the page would you like to see updated?

The `warn_error` config (`--warn-error` flag, `DBT_WARN_ERROR` env var) will remain a simple boolean: treat all warnings as errors, or none.

Users can choose the more granular `warn_error_options` config (`--warn-error-options`, `DBT_WARN_ERROR_OPTIONS` env var) to specify a subset of warnings to treat as errors, via an `include`/`exclude` syntax that we're likely to reuse elsewhere.

The names of warnings are determined based on the names of `WarnLevel` events here: https://github.com/dbt-labs/dbt-core/blob/main/core/dbt/events/types.py

This is a configuration intended for intermediate/advanced users. Specific use cases we've seen requested in past: https://github.com/dbt-labs/dbt-core/issues/4383, https://github.com/dbt-labs/dbt-core/issues/6080

### Additional information

https://github.com/dbt-labs/dbt-core/issues/6165
https://github.com/dbt-labs/dbt-core/pull/6520

### PR DESCRIPTION
**Pull request by MichelleArk**
## What are you changing in this pull request and why?
fixes: https://github.com/dbt-labs/docs.getdbt.com/issues/2687

## Checklist

- [x] Review the [Content style guide](https://github.com/dbt-labs/docs.getdbt.com/blob/current/contributing/content-style-guide.md) and [About versioning](https://github.com/dbt-labs/docs.getdbt.com/blob/current/contributing/single-sourcing-content.md#adding-a-new-version) so my content adheres to these guidelines.
- [x] Add a checklist item for anything that needs to happen before this PR is merged, such as "needs technical review" or "change base branch."

### CODE REVIEW

File: website/docs/reference/global-configs.md [hunk 1]
```diff
@@ -405,7 +405,34 @@ Turning on the `WARN_ERROR` config will convert dbt warnings into errors. Any ti
...
+...
+```
+
+```
+config:
+  warn_error_options:
+    include: all
+    exclude: 
+      - NoNodesForSelectionCriteria
+```
```
Comment (jtcohen6) [inline website/docs/reference/global-configs.md]: If you want to make it prettier:
```
<File name='profiles.yml'>
```yaml
config:
  warn_error_options:
    include: all
    exclude: 
      - NoNodesForSelectionCriteria
<replace this line with backticks: ```>
</File>
```

File: website/docs/reference/profiles.yml.md [hunk 2]
```diff
@@ -18,6 +18,7 @@ This article lists the parts of your `profiles.yml` which are _not_ database spe
   [printer_width](global-configs#printer-width): <integer>
   [write_json](global-configs#writing-json-artifacts): <true | false>
   [warn_error](global-configs#warnings-as-errors): <true | false>
+  [warn_error_options](global-configs#warnings-as-errors-options): < include: all | include: [<error-name>] | include: all, exclude: [<error-name>] >
```
Comment (jtcohen6) [inline website/docs/reference/profiles.yml.md]: This renders nicely - nice guesswork (??)

Small thing - agree with not adding a new heading to the other page, but discussing both configs under one heading - so we just need to keep the same anchor:
```suggestion
  [warn_error_options](global-configs#warnings-as-errors): < include: all | include: [<error-name>] | include: all, exclude: [<error-name>] >
```

File: website/docs/reference/global-configs.md [hunk 3]
```diff
@@ -405,7 +405,34 @@ Turning on the `WARN_ERROR` config will convert dbt warnings into errors. Any ti
 $ dbt --warn-error run
 ...
 ```
+</File>
+
+Turning on the WARN_ERROR_OPTIONS config will convert specific dbt warnings into errors. Warnings that should be converted into errors can be specified through ‘include’ and/or ‘exclude’ parameters. Warning names can be found in [https://github.com/dbt-labs/dbt-core/blob/main/core/dbt/events/types.py](dbt-core's types.py file), where each class name that inherits from `WarnLevel` corresponds to a warning name (ex: `AdapterDeprecationWarning`, `NoNodesForSelectionCriteria`).
```
Comment (jtcohen6) [inline website/docs/reference/global-configs.md]: Suggest rewording slightly to tie into the previous config, and indicate who this is for. ("Has this ever happened to you? You're an intermediate/advanced user of dbt, you love the `--warn-error` config, but there are some warnings you just don't care about, or some you care about a *lot*. This config gives you more granular control over exactly which types of warnings are treated as errors...")

Need to also note here that `WARN_ERROR` and `WARN_ERROR_OPTIONS` are mutually exclusive—you can only specify one, even when you're specifying the config in multiple places (e.g. env var + flag), otherwise you'll see an error

Bonus nits:
- link text needs to switch order: `[dbt-core's types.py file](https://github.com/dbt-labs/dbt-core/blob/main/core/dbt/events/types.py)`
- I'd put backticks around `include` + `exclude` since they represent code that users will be writing
- I prefer `e.g.` over `ex.`, looks like [experts agree](https://preply.com/en/question/example-ex-or-eg-41620)

File: website/docs/reference/global-configs.md [hunk 4]
```diff
@@ -405,7 +405,34 @@ Turning on the `WARN_ERROR` config will convert dbt warnings into errors. Any ti
...
+
+<File name='Usage'>
+
+```text
+$ dbt --warn-error-options ‘{“include”: “all”}’ run
```
Comment (jtcohen6) [inline website/docs/reference/global-configs.md]: nit: replace `‘` + `’` with `'`, and `“` + `”` with `"` (unless you feel strongly?)

File: website/docs/reference/global-configs.md [hunk 5]
```diff
@@ -405,7 +405,34 @@ Turning on the `WARN_ERROR` config will convert dbt warnings into errors. Any ti
...
+...
+```
 
+```text
+$ dbt --warn-error-options ‘{“include”: “all”, “exclude”:[NoNodesForSelectionCriteria]}’ run
+...
+```
+
+```text
+$ dbt --warn-error-options ‘{“include”: [NoNodesForSelectionCriteria]}’ run
```
Comment (jtcohen6) [inline website/docs/reference/global-configs.md]: One of these examples could also use the `DBT_WARN_ERROR_OPTIONS` env var. (That's how dbt Cloud users are most likely to set it).

**Review by jtcohen6 — COMMENTED** [review]
Thanks for taking the initiative on this! I'm happy to pick this up & take over the finish line if you'd rather hand it off - jlmk

Once the v1.4 migration guide is initialized (https://github.com/dbt-labs/docs.getdbt.com/pull/2646), we'll also want a note in there linking back to this update. I can hit that in a follow-up PR, to avoid lots of merge conflicts.

File: website/docs/reference/profiles.yml.md (hunk 2 above)
Comment (MichelleArk) [reply website/docs/reference/profiles.yml.md]: good eye!

File: website/docs/reference/global-configs.md (hunk 4 above)
Comment (MichelleArk) [reply website/docs/reference/global-configs.md]: oops. this is what I get for using apple notes as my scratchpad

Comment (jtcohen6) [reply website/docs/reference/global-configs.md]: the apple notes > docs PR > copy-paste > `dbt: error: ... invalid choice: '“all”'` pipeline

**Review by jtcohen6 — DISMISSED** [review]
you nailed it!

File: website/docs/reference/global-configs.md [hunk 6]
```diff
@@ -405,7 +405,50 @@ Turning on the `WARN_ERROR` config will convert dbt warnings into errors. Any ti
 $ dbt --warn-error run
 ...
 ```
+</File>
+
```
Comment (jtcohen6) [inline website/docs/reference/global-configs.md:409]: Ah, sorry, one last thing - because this feature is new in v1.4, let's just wrap everything below in a ["VersionBlock"](https://github.com/dbt-labs/docs.getdbt.com/blob/current/contributing/single-sourcing-content.md#versioning-blocks-of-content):
```
<VersionBlock firstVersion="1.4">
... everything you're adding below ...
</VersionBlock>
```

That will work once v1.4 is added to the set of site-wide ["dbt versions,"](https://github.com/dbt-labs/docs.getdbt.com/blob/current/website/dbt-versions.js#L1) which Matt is tackling in https://github.com/dbt-labs/docs.getdbt.com/pull/2534.

Unfortunately, the `VersionBlock` rendering doesn't yet work _within_ code blocks, so no need to do this for the small change to `profiles.yml.md`.

Comment (MichelleArk) [reply website/docs/reference/global-configs.md:409]: cooooool!

**Review by jtcohen6 — APPROVED** [review]
